Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -0,0 +1,306 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Remotion.ObjectBinding.Validation;
+using Remotion.Utilities;
+
+namespace Remotion.ObjectBinding.Web.UI.Controls.Validation
+{
+  public class BocListValidationFailureRepository
+  {
+    private class HandleableFailure
+    {
+      public BusinessObjectValidationFailure Failure { get; }
+
+      public bool Handled { get; private set; }
+
+      public HandleableFailure (BusinessObjectValidationFailure failure)
+      {
+        Failure = failure;
+        Handled = false;
+      }
+
+      public void MarkAsHandled ()
+      {
+        Handled = true;
+      }
+    }
+
+    private class RowFailureContainer
+    {
+      private List<HandleableFailure>? _rowFailures;
+      private Dictionary<IBocColumnDefinitionWithValidationSupport, List<HandleableFailure>>? _cellFailures;
+
+      public void AddRowFailures (IEnumerable<BusinessObjectValidationFailure> failures)
+      {
+        _rowFailures ??= new List<HandleableFailure>();
+        _rowFailures.AddRange(failures.Select(e => new HandleableFailure(e)));
+      }
+
+      public void AddCellFailures (IBocColumnDefinitionWithValidationSupport cellDefinition, IEnumerable<BusinessObjectValidationFailure> failures)
```

Comment:
the api supports adding extra validation failures to an existing item. if we need it, this is okay. I don't know it's needed.

Reply:
I don't think the validators should be required to add all the failures beforehand so that one validator does not remove the failures input by another validator. 

Therefore, I think we should leave this in here, just because it makes using the repository much more failsafe.